src/operators/basic_operators.jl

```diff
@@ -86,9 +86,11 @@ struct DiffEqArrayOperator{T,AType<:AbstractMatrix{T},F} <: AbstractDiffEqLinear
     new{eltype(A),AType,typeof(update_func)}(A, update_func)
 end
 
+has_adjoint(::DiffEqArrayOperator) = true
```

Comment:
add `has_adjoint(::AbstractMatrix) = true` as well. That's defined in the abstract matrix interface that it can be done via index swapping.